pkg/storage/util.go

```diff
@@ -56,6 +58,10 @@ func NamespaceKeyFunc(prefix string, obj runtime.Object) (string, error) {
 	if err != nil {
 		return "", err
 	}
+	name := meta.Name()
+	if ok, msg := validation.ValidatePathSegmentName(name, false); !ok {
```

Comment:
hypothetically.  I'm saying I'm OK asserting that any objects we create will be a subdomain or stricter.  But it's not a huge deal, I guess.